lib/datautils/owid/datautils/dataframes.py

```diff
@@ -461,6 +461,13 @@ def map_series(
         # Replace those nans with their original values, except if they were actually meant to be mapped to nan.
         # For example, if {"bad_value": np.nan} was part of the mapping, do not replace those nans back to "bad_value".
 
+        # if we are setting values from the original series, ensure we have the same dtype
+        try:
+            series_mapped = series_mapped.astype(series.dtype, copy=False)
```

Comment:
This might be specific to a corner case in the tests, but essentially what happened here is that if the mapping does not have anything that is in the original `series`, the resulting `series_mapped` ends up being a Series with all NaNs, and thus gets inferred to have a float dtype.  
And then a bit below you do `series_mapped.loc[missing] = series[missing]`, assigning values from the original series into the result. Starting with pandas 3.0, such a setitem operation will preserve the dtype of the series on the left, i.e. a float here because of all-NaN. And at that point, assigning strings into that will give an error (it gives a warning right now with pandas 2)